app/models/service_template_ansible_playbook.rb

```diff
@@ -1,6 +1,4 @@
 class ServiceTemplateAnsiblePlaybook < ServiceTemplateGeneric
-  before_destroy :check_retirement_potential
```

Comment:
we should still have `before_destroy`, but with  `:prepend => true`, which will not delete association at the time when the call back happens.
```
  before_destroy :check_retirement_potential, :prepend => true
```